**Analyzing the dataset**

This python notebook analyzes the `housing.csv` dataset to determine the various transformation to be done to transform the data into a format suitable for training the machine learning model (linear regression).

In [3]:
import pandas as pd
import numpy as np

In [5]:
dataset = pd.read_csv("housing.csv", parse_dates=True)
row_count = dataset.shape[0]
column_count = dataset.shape[1]
print(f"Dataset has {row_count} row(s) and {column_count} column(s).")
print("Information about the dataset is displayed below:")
print(dataset.info())

Dataset has 545 row(s) and 13 column(s).
Information about the dataset is displayed below:
<class 'pandas.core.frame.DataFrame'>
RangeIndex: 545 entries, 0 to 544
Data columns (total 13 columns):
 #   Column            Non-Null Count  Dtype 
---  ------            --------------  ----- 
 0   price             545 non-null    int64 
 1   area              545 non-null    int64 
 2   bedrooms          545 non-null    int64 
 3   bathrooms         545 non-null    int64 
 4   stories           545 non-null    int64 
 5   mainroad          545 non-null    object
 6   guestroom         545 non-null    object
 7   basement          545 non-null    object
 8   hotwaterheating   545 non-null    object
 9   airconditioning   545 non-null    object
 10  parking           545 non-null    int64 
 11  prefarea          545 non-null    object
 12  furnishingstatus  545 non-null    object
dtypes: int64(6), object(7)
memory usage: 55.5+ KB
None


In [10]:
print(f"The maximum and minimum price values are ${dataset['price'].max():,} and ${dataset['price'].min():,} respectively.")
print(f"The maximum and minimum area are {dataset['area'].max():,} and {dataset['area'].min():,} respectively.")

The maximum and minimum price values are $13,300,000 and $1,750,000 respectively.
The maximum and minimum area are 16200 and 1650 respectively.


**Observations:**

- All 545 rows have non-null values and so null-cleanup is not required.
- Since the price value in the dataset is atleast 1.75 millions, dividing all the values in the dataset by 1 million will give us a relatively lesser scale of values to train our model with, making it less computabionally intensive.
- Since the area is atlease 1650, we can divide all the area values by 1000, to simplify computations.